src/modules/todos/todos.service.ts

```diff
@@ -0,0 +1,119 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { TodoEntity } from 'src/infrastructure/entity/todo.entity';
+import { Repository } from 'typeorm';
+import { CreateTodoResponse, GetTodosResponse } from './graphql/todos.response';
+import { ITodoList } from './dto/todos.dto';
+import { CreateTodoInput, RemoveTodoInput } from './graphql/todos.input';
+import { ProjectEntity } from 'src/infrastructure/entity/projects.entity';
+import { SYSTEM } from 'src/constants';
+
+@Injectable()
+export class TodosService {
+  constructor(
+    @InjectRepository(TodoEntity)
+    private readonly todoRepo: Repository<TodoEntity>,
+    @InjectRepository(ProjectEntity)
+    private readonly projectRepo: Repository<TodoEntity>,
+  ) {}
+
+  async getTodos(): Promise<GetTodosResponse> {
+    try {
+      const todoData = await this.todoRepo.find({
+        order: { updatedDateTime: 'DESC' },
+      });
+      if (todoData.length === 0) throw new Error('No Todo Found!');
+      const todoArray: ITodoList[] = todoData.map(
+        ({ id, title, description, priority, isDone, dueDate, remarks }) => ({
+          id,
+          title,
+          description,
+          priority,
+          isDone,
+          dueDate,
+          remarks,
+        }),
+      );
+      return { result: todoArray };
+    } catch (error) {
+      console.log(error);
+      throw new Error(error);
+    }
+  }
+
+  async createTodo(input: CreateTodoInput): Promise<CreateTodoResponse> {
+    try {
+      const { id, title, description, priority, isDone, dueDate, remarks } =
+        input;
+
+      // Validation the id of the project
+      const projectData = await this.projectRepo.findOneBy({ id });
+      if (!projectData)
+        throw new Error('No project available. Please contact support!');
+
+      // Then add the record
+      const newTodo = new TodoEntity();
+      newTodo.createdBy = SYSTEM;
+      newTodo.title = title;
+      newTodo.description = description;
```

Comment:
Since description, dueDate and remarks are optional. If the user didnt add in the input, let show '-'